lib/calendar.ts

```diff
@@ -1099,7 +1148,10 @@ const helperIslamic = ObjectAssign({}, nonIsoHelperBase, {
   calendarType: 'lunar',
   inLeapYear(calendarDate, cache) {
     // In leap years, the 12th month has 30 days. In non-leap years: 29.
-    const days = this.daysInMonth({ year: calendarDate.year, month: 12, day: 1 }, cache);
+    const days = (this as unknown as NonIsoHelperBase).daysInMonth(
```

Comment:
Instead of these casts, would it make sense to put a `this: NonIsoHelperBase` into the param declaration like you did in other files?

Reply:
Done - replaced all the `as unknown as` with the equivalent `this` type.